lib/src/components/NxToast/types.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import { HTMLAttributes, ReactNode, ReactElement } from 'react';
+import * as PropTypes from 'prop-types';
+
+// import { allToastTypes, ToastType } from '../../util/toastLevels';
+
+export type Props = HTMLAttributes<HTMLDivElement>;
+
+export const propTypes = {
+  className: PropTypes.string,
+  onClose: PropTypes.func
+} as PropTypes.ValidationMap<Props>;
+
+export const nxToastPropTypes: PropTypes.ValidationMap<NxToastProps> = {
+  toastId: PropTypes.number.isRequired,
+  onClose: PropTypes.func.isRequired,
+  children: PropTypes.any.isRequired,
+  previousFocusedElement: PropTypes.any
+};
+
+export interface NxToastProps {
+  onClose: () => void;
+  toastId: number;
+  children: ReactElement;
+  previousFocusedElement?: ReactNode;
+}
+
+export type ToastModel = {
+  toastId: number;
+};
+
+export type FocusContextType = {
+  adjustFocus: ()=> void;
```

Comment:
I think it would decrease coupling if you changed the names here to not actually be about focus.  The purpose of the context is to establish a line on communication between the toasts and the toast container.  So let's call it `NxToastContainerContext`.  And the only thing that needs to be communicated between the two is when a toast is closing, so lets call the function `onToastClosing`.  That the container will do a bunch of focus stuff in response to that function being called is its internal business, not something that `NxToast` itself needs to know about.

Reply:
> I think it would decrease coupling if you changed the names here to not actually be about focus. The purpose of the context is to establish a line on communication between the toasts and the toast container. So let's call it `NxToastContainerContext`. And the only thing that needs to be communicated between the two is when a toast is closing, so lets call the function `onToastClosing`. That the container will do a bunch of focus stuff in response to that function being called is its internal business, not something that `NxToast` itself needs to know about.

Renamed to NxToastContainerContext + onToastClosing here: [6d9117f](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/6d9117f2dd7cd5cf68d865be71da24f5cca7c965)